components/content-service/pkg/service/headless-log-service.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package service
+
+import (
+	"context"
+	"strings"
+
+	"github.com/opentracing/opentracing-go"
+	"golang.org/x/xerrors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/common-go/tracing"
+	"github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/content-service/pkg/logs"
+	"github.com/gitpod-io/gitpod/content-service/pkg/storage"
+)
+
+// HeadlessLogService implements LogServiceServer
+type HeadlessLogService struct {
```

Comment:
unit tests for the `HeadlessLogService` would be awfully nice  - e.g. by mocking `PresignedAccess`

Reply:
I added a test for `ListLogs`. Not super convinced that the test alone justifies the LOC of the mock... but maybe useful for the other services as well.